journals/apps/journals/models.py

```diff
@@ -469,12 +476,54 @@ def get_context(self, request, *args, **kwargs):
             context['user_has_access'] = JournalAccess.user_has_access(request.user, self.journal)
         else:
             context['user_has_access'] = False
-        discovery_journal_api_client = self.site.siteconfiguration.discovery_journal_api_client
-        journal_data = discovery_journal_api_client.journals(self.journal.uuid).get()
+        journal_data = self._get_journal_from_discovery()
         context['journal_data'] = journal_data
         context['buy_button_url'] = self.generate_require_auth_basket_url(journal_data['sku'])
         return context
 
+    def _get_journal_from_discovery(self):
+        '''
+        Get journal data from discovery first
+        checking in cache
+        '''
+        api_resource = 'journals'
+
+        cache_key = get_cache_key(
+            resource=api_resource,
+            journal_uuid=self.journal.uuid,
+            journal_about_id=self.id,
+        )
+
+        _CACHE_MISS = object()
+        journal_data = cache.get(cache_key, _CACHE_MISS)
+
+        if journal_data is _CACHE_MISS:
+            api_client = self.site.siteconfiguration.discovery_journal_api_client
+            journal_data = api_client.journals(self.journal.uuid).get()
+            cache.set(cache_key, journal_data, 3600)
```

Comment:
Might want to verify the API call was successful before we cache it for an hour.